src/librustc_typeck/impl_wf_check.rs

```diff
@@ -99,6 +99,10 @@ fn enforce_impl_params_are_constrained(
 ) {
     // Every lifetime used in an associated type must be constrained.
     let impl_self_ty = tcx.type_of(impl_def_id);
+    if impl_self_ty.sty == ty::Error {
+        // Don't complain about unconstrained type params when self ty doesn't exist. (#36836)
+        return;
```

Comment:
"doesn't exist" isn't the best description. It's more like "isn't known (because of errors)".
One way to be sure it came from an error is to add a `tcx.sess.delay_span_bug` call (i.e. turns into an ICE if there isn't an error elsewhere).